From 925fb717b151d082780020504a55fe01d2f5b960 Mon Sep 17 00:00:00 2001
From: Ramneek Mehresh <ramneek.mehresh@freescale.com>
Date: Thu, 11 Apr 2013 17:07:31 +0530
Subject: [PATCH 363/741] usb: host: Remove host drv upon otg bring-up

commit db210aa36dafd77e63640265c8fc50d73e3d71fd from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

Change have_hcd variable to remove/suspend host driver on
completion of otg initialization for otg auto detect

Signed-off-by: Ramneek Mehresh <ramneek.mehresh@freescale.com>
Signed-off-by: yinbo.zhu <yinbo.zhu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/usb/host/ehci-fsl.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/usb/host/ehci-fsl.c b/drivers/usb/host/ehci-fsl.c
index 3b76f870d2fd..0a3dbd256d91 100644
--- a/drivers/usb/host/ehci-fsl.c
+++ b/drivers/usb/host/ehci-fsl.c
@@ -219,6 +219,7 @@ static int fsl_ehci_drv_probe(struct platform_device *pdev)
 #if defined(CONFIG_FSL_USB2_OTG) || defined(CONFIG_FSL_USB2_OTG_MODULE)
 	if (pdata->operating_mode == FSL_USB2_DR_OTG) {
 		struct ehci_hcd *ehci = hcd_to_ehci(hcd);
+		struct ehci_fsl *ehci_fsl = hcd_to_ehci_fsl(hcd);
 
 		hcd->usb_phy = usb_get_phy(USB_PHY_TYPE_USB2);
 
@@ -241,6 +242,10 @@ static int fsl_ehci_drv_probe(struct platform_device *pdev)
 		}
 
 		hcd->skip_phy_initialization = 1;
+		ehci_fsl->have_hcd = 1;
+	} else {
+		dev_err(&pdev->dev, "wrong operating mode\n");
+		return -ENODEV;
 	}
 #endif
 	return retval;
-- 
2.17.1

